lib/DDGC/Util/Script/Docs.pm

```diff
@@ -0,0 +1,283 @@
+
+            return unless $name =~ /^[^.].+\.md$/;    # only markdown files
+
+            my ($file) = $name =~ /^$self->{source_dir}\/(.*)\.md/;
+            my $dir_rel = '';
+            ( $dir_rel, $file ) = $file =~ /^(.*)\/(.*)/ if $file =~ /\//;
+
+            my $markdown = io($name)->slurp;
+
+# Replaces hard github links to other markdown files to our newly converted relative links
+            $markdown =~
+s~(\]\()https://github.com/duckduckgo/duckduckgo-documentation/blob/master/duckduckhack/(?:[^\/\.]+\/){1,4}([^\.]+?)\.md([^\)]*?\))~$1$2$3~sg;
+
+ # Replace code types the parser can't deal with,
+ # that is a #### line without a space line between it and a code start line ```
+            $markdown =~ s/(\n\#+.*?)(\n`)/$1\n$2/sg;
+
+            my $buffer = q{};
+            open my $fh, '>', \$buffer;
+
+            my $handler = Markdent::Handler::HTMLStream::Document->new(
+                title  => $name,
+                output => $fh,
+            );
+
+            my $parser =
+              Markdent::Parser->new( dialect => 'GitHub', handler => $handler );
+            $parser->parse( markdown => $markdown );
+
+            my $html = $buffer;
+
+            $html =~ s/^.*?<html>.*?<\/head>\s*<body>\s*//s;
+            $html =~ s/\s*<\/body>.*?<\/html>\s*$//s;
+
+            $html =~
+s/(<h\d>)(.*?)(<\/h\d>)/$1 . '<a name="' . make_anchor($2) . '" class="anchor"><\/a>' . $2 . $3/ges;
+
+            my $category = $nav{$file}{'category'} || '';
+            my $title    = $nav{$file}{'title'}    || '';
+
+            warn qq(NO CATEGORY: $file\n) if !$category;
+
+            my $prev = $prev_next{$file}{'prev'} || [];
+            my $next = $prev_next{$file}{'next'} || [];
+
+            warn qq(NO PREV: $file\n) if !$prev;
+            warn qq(NO NEXT: $file\n) if !$next;
+
+            $pages{$file} = {
+                file       => $file,
+                  file_dir => $dir_rel,
+                  file_path =>
+"https://github.com/duckduckgo/duckduckgo-documentation/tree/master/duckduckhack",
+                  title        => $title,
+                  category     => $category,
+                  html         => $html,
+                  nav_ref      => $nav_arr_ref,
+                  maintemplate => 'doc.tx',
+                  prev         => $prev,
+                  next         => $next,
+            };
+
+        },
+        $self->source_dir
+    );
+
+    return \%pages;
+}
+
+sub make_anchor {
+    my ($anchor) = @_;
+
+    $anchor = lc $anchor;
+    $anchor =~ s/[^a-z\s]+//g;
+    $anchor = trim $anchor;
+    $anchor =~ s/\s+/\-/g;
+
+    return $anchor;
+}
+
+sub execute {
+    my ( $self ) = @_;
+    my $tmp = tempdir;
+    my $error = 0;
+
+    try {
+        my $pages = $self->pages;
+
+        for my $file ( keys $pages ) {
+            my $page = $pages->{$file};
+
+            next if !$page->{maintemplate};
+
+            $self->xslate->render(
```

Comment:
I see what you mean now (I was reading the lines above, not underneath - wonder if this comment tool can be made better). This is not robust anyway.

Jason's fix is a bit more explicit, I think.